easybuild/easyconfigs/s/Salmon/Salmon-1.10.3-GCC-12.3.0.eb

```diff
@@ -0,0 +1,59 @@
+# #
+# This is a contribution from DeepThought HPC Service, Flinders University, Adelaide, Australia
+# Homepage:     https://staff.flinders.edu.au/research/deep-thought
+#
+# Authors::     Robert Qiao <rob.qiao@flinders.edu.au>
+# License::     GLPv2
+#
+# Notes::
+# #
+# Contribution from the NIHR Biomedical Research Centre
+# Guy's and St Thomas' NHS Foundation Trust and King's College London
+# uploaded by J. Sassmannshausen
+# Thanks to people like Simon Brandord, Mikael Oehmann and Kenneth Hoste
+# for their help with the ICE
+
+easyblock = 'CMakeMake'
+
+name = 'Salmon'
+version = '1.10.3'
+
+homepage = 'https://github.com/COMBINE-lab/salmon'
+description = """Salmon is a wicked-fast program to produce a highly-accurate,
+ transcript-level quantification estimate from RNA-seq data."""
+
+toolchain = {'name': 'GCC', 'version': '12.3.0'}
+toolchainopts = {'pic': True}
+
+source_urls = ['https://github.com/COMBINE-lab/salmon/archive']
+sources = [{'download_filename': 'v%(version)s.tar.gz', 'filename': SOURCE_TAR_GZ}]
+checksums = ['a053fba63598efc4ade3684aa2c8e8e2294186927d4fcdf1041c36edc2aa0871']
```

Comment:
@pavelToman is there a specific reason why you've changed the way the source download is specified?

Reply:
If I got it @boegel ask me to make `source` like this: https://github.com/easybuilders/easybuild-easyconfigs/pull/20852#discussion_r1705629115

Reply:
I don't disagree, just wasn't aware of the change in policy, if that's what it is.
@boegel?

Reply:
renaming is not needed here as there's only 1 source tarball (no confusion possible), but it doesn't hurt